File: src/main/java/com/dashboard/interfaces/serviceInterfaces/TaskServiceInterface.java
package com.dashboard.interfaces.serviceInterfaces;

import com.dashboard.models.JPA.Priority;
import com.dashboard.models.JPA.Task;

import java.util.List;

public interface TaskServiceInterface {

    void deleteAll();

    void addAllTasks(List<Task> tasks);

    Long totalEmployees();

    List<Task> getTaskByPriority(Priority priority);

    List<Task> findTaskByStatus(String status);

    long count();
}

File: src/main/kotlin/dashboard/interfaces/serviceInterfaces/TaskServiceInterface.kt
package dashboard.interfaces.serviceInterfaces

import dashboard.models.jpa.Priority
import dashboard.models.jpa.Task

interface TaskServiceInterface {
    fun deleteAll()
    fun addAllTasks(tasks: List<Task?>?)
    fun totalEmployees(): Long?
    fun getTaskByPriority(priority: Priority?): List<Task?>?
    fun findTaskByStatus(status: String?): List<Task?>?
    fun count(): Long
}